cmd/version/cmd.go

```diff
@@ -17,77 +17,104 @@ limitations under the License.
 package version
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 
 	verify "github.com/openshift/rosa/cmd/verify/rosa"
 	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/reporter"
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var (
-	args struct {
-		clientOnly bool
-		verbose    bool
+var args struct {
```

Comment:
Yeah sounds good, will update that

Reply:
Can you elaborate on this, as we need the args as a var to pass to the flags

Reply:
I was thinking if we can get rid of the global args variable as it feels a bit out-of-place that everything else is well encapsulated. 

Open to suggestions here, but I was thinking something like this:

Define a struct for the options the user can pass with a function to create a new, default set of these options:

```
type RosaVersionUserOptions struct {
	clientOnly bool
	verbose    bool
}

func NewRosaVersionUserOptions() RosaVersionUserOptions {
	return RosaVersionUserOptions{}
}
```

These options are then bound to the flagset and passed into the Runner, like so:

```
func NewRosaVersionCommand() *cobra.Command {

	options := NewRosaVersionUserOptions()

	cmd := &cobra.Command{
		Use:   use,
		Short: short,
		Long:  long,
		Args:  cobra.NoArgs,
		Run:   rosa.DefaultRunner(rosa.RuntimeWithOCM(), RosaVersionRunner(options)),
	}

	cmd.Flags().SortFlags = false
	cmd.Flags().BoolVar(
		&options.clientOnly,
		"client",
		false,
		"Client version only (no remote version check)",
	)
	cmd.Flags().BoolVarP(
		&options.verbose,
		"verbose",
		"v",
		false,
		"Display verbose version information, including download locations",
	)
	return cmd
}
```

Finally, the `RosaVersionOptions` has a method that lets it bind from the user options e.g:

```
func (o *RosaVersionOptions) BindAndValidate(options RosaVersionUserOptions) {
	o.verbose = options.verbose
	o.clientOnly = options.clientOnly
}
```

and the runner is called like this:

```
func RosaVersionRunner(userOptions RosaVersionUserOptions) rosa.CommandRunner {
	return func(ctx context.Context, runtime *rosa.Runtime, command *cobra.Command, args []string) error {
		options, err := NewRosaVersionOptions()
		options.BindAndValidate(userOptions)
		if err != nil {
			return fmt.Errorf("there was a problem creating version options: %v", err)
		}
		return options.Version()
	}
}
```

In that model, the runner expects a set of user options, but how we bind the values into it...not a concern of the runner. For testing we just pass in the struct directly, otherwise we lean on Cobra to do it for us.

WDYT?

Reply:
Yeah sounds good to me, @hunterkepley can you take a look at this pattern and we follow the same in your refactor of the describe machine pool